test/liquidity-curve.test.js

```diff
@@ -132,15 +132,33 @@ describe('LiquidityCurve', function () {
       assert.equal(combinedCurve.amountAt(70), 70)
     })
 
-    it('cleans up duplicate points', function () {
+    it('doesn\'t generate duplicate points', function () {
       const curve1 = new LiquidityCurve([ [1, 0], [50000000001, 49800199999], [100000000000001, 49800199999] ])
       const curve2 = new LiquidityCurve([ [2, 0], [50000000001, 49800199999], [100000000000001, 49800199999] ])
       const combinedCurve = curve1.combine(curve2)
-
       assert.deepStrictEqual(combinedCurve.getPoints(),
         [ [1, 0], [2, 0], [50000000001, 49800199999], [100000000000001, 49800199999] ])
     })
 
+    it('cleans up duplicate points caused by rounding', function () {
+      const curve1 = new LiquidityCurve([ [0, 0], [2, 2] ])
+      const curve2 = new LiquidityCurve([ [1, 0], [2, 9] ])
+      const combinedCurve = curve1.combine(curve2)
+
+      // Before rounding, the curve is:
+      // [ [0,0], [1,1], [1.125,1.125], [2,9] ]
+      assert.deepStrictEqual(combinedCurve.getPoints(),
+        [ [0, 0], [1, 1], [2, 9] ])
+
+      const curve3 = new LiquidityCurve([ [0, 0], [2000000, 2000000] ])
+      const curve4 = new LiquidityCurve([ [1999999, 0], [2000000, 9000000] ])
+      const combinedCurve2 = curve3.combine(curve4)
+      // Before rounding, the curve is:
```

Comment:
'Due to rounding' instead of 'Before rounding'. Rounding is what caused these duplicate integers.